src/generic/AbsMSeries.jl

```diff
@@ -92,9 +92,9 @@ truncated to the new precisions. The mutated series is returned.
 """
 function set_precision!(a::AbsMSeries, prec::Vector{Int})
     parent(a).weighted_prec != -1 && error("Operation not possible in weighted rings")
-    length(prec) != length(precision(a)) &&
+    length(prec) != length(a.prec) &&
                          error("Array length not equal to number of variables")
-    if !exponents_lt(precision(a), prec)
+    if !exponents_lt(a.prec, prec)
```

Comment:
Does this function not exist or what is the problem?

Reply:
This is because `precision(a)` is not type stable: it can return either an `Int` or a `Vector{Int}`:
```julia
@doc raw"""
    precision(a::AbsMSeries)

Return a vector of precisions, one for each variable in the series ring.
If the ring is weighted the weighted precision is returned instead.
"""
function precision(a::AbsMSeries)
   S = parent(a)
   if S.weighted_prec == -1
      return a.prec            # Vector{Int}
   else
      return S.weighted_prec   # Int
   end
end
```
As a result we have code looking like the one you are commenting on here, i.e.:
```
    parent(a).weighted_prec != -1 && error("Operation not possible in weighted rings")
    length(prec) != length(precision(a)) &&
                         error("Array length not equal to number of variables")
    if !exponents_lt(precision(a), prec)
        a.poly = truncate_poly(a.poly, prec)
    end
```
The change above was the simples fix I can think of, but IMHO the real problem is the type instability.

But I don't understand enough about what this code is supposed to do to even make a proposal... to me it seems just wrong to do that (sometimes return `Int`, sometimes `Vector{Int}`), but perhaps there are great reasons why it is best this way?

Maybe also @fieker has comments on this?

Anyway, I can happily move this part of the PR to a separate one

Reply:
Yeah, that does not look great. Thanks for the explanation.

Reply:
I moved this into a separate PR (and extended it to really resolve all JET warnings in that file).